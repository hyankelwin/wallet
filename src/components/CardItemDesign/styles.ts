import styled from 'styled-components';
import Color from '../../styles/color';

interface Image {
  src: string;
}

export const Container = styled.div`
  margin: 4px;

  div:hover {
    transform: scale(1.3);
    transition: ease-in;
    transition-duration: 150ms;
  }
`;

export const Content = styled.div<Image>`
  border: 1px solid ${Color.grey};
  border-radius: 5px;
  width: 100px;
  height: 100px;
  background: ${props => `url(${props.src})`};
`;
