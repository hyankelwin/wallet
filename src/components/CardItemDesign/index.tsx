import React from 'react';
import { Container, Content } from './styles';

interface CardItemDesignProps {
  src: string;
}

const CardItemDesign: React.FC<CardItemDesignProps> = ({ src }) => {
  return (
    <Container>
      <Content src={src} />
    </Container>
  );
};

export default CardItemDesign;
