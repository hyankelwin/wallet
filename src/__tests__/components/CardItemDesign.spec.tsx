import { mount } from 'enzyme';
import CardItemDesign from '../../components/CardItemDesign';

describe('CardItemDesign component', () => {
  it('should be able to render CardItemDesign component with success', () => {
    const img = '/static/logo.png';
    const wrapper = mount(<CardItemDesign src={img} />);

    expect(wrapper.debug()).toMatchSnapshot();
  });
});
