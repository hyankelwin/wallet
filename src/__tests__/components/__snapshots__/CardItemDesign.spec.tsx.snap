// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardItemDesign component should be able to render CardItemDesign component with success 1`] = `
"<CardItemDesign src=\\"/static/logo.png\\">
  <styled.div>
    <div className=\\"sc-bdvvtL cRHldS\\">
      <styled.div src=\\"/static/logo.png\\">
        <div src=\\"/static/logo.png\\" className=\\"sc-gsDKAQ braFDF\\" />
      </styled.div>
    </div>
  </styled.div>
</CardItemDesign>"
`;
